mars/learn/contrib/pytorch/dataset.py

```diff
@@ -48,14 +59,14 @@ def __getitem__(self, index):
     @staticmethod
     def get_data(t, index):
         if isinstance(t, TENSOR_TYPE):
-            return t[index].execute().fetch()
+            return t[index].fetch()
         elif isinstance(t, np.ndarray):
             return t[index]
         elif isinstance(t, DATAFRAME_TYPE):
-            return t.iloc[index].execute().fetch().values
+            return t.iloc[index:index+1].fetch().values[0]
```

Comment:
The original version is `return t.iloc[index].fetch().values`. It will get a Series data and fetch() to get np.ndarray data. But it got an error `AssertionError: series name in metadata None is not equal to real name 1`. The details list below. Now this version can work well. 

dataset.py:57: in __getitem__
    return tuple(self.get_data(t, index) for t in self._tileables)
dataset.py:57: in <genexpr>
    return tuple(self.get_data(t, index) for t in self._tileables)
dataset.py:69: in get_data
    print(type(t.iloc[index].fetch()))
../../../dataframe/core.py:525: in fetch
    return self._fetch(session=session, **kw)
../../../core/entity/executable.py:122: in _fetch
    return fetch(self, session=session, **kw)
../../../deploy/oscar/session.py:1393: in fetch
    return session.fetch(tileable, *tileables, **kwargs)
../../../deploy/oscar/session.py:1242: in fetch
    return asyncio.run_coroutine_threadsafe(coro, self._loop).result()
/home/yuanchong/.conda/envs/mars-dev/lib/python3.7/concurrent/futures/_base.py:435: in result
    return self.__get_result()
/home/yuanchong/.conda/envs/mars-dev/lib/python3.7/concurrent/futures/_base.py:384: in __get_result
    raise self._exception
../../../deploy/oscar/session.py:1377: in _fetch
    data = await session.fetch(tileable, *tileables, **kwargs)
../../../deploy/oscar/tests/session.py:80: in fetch
    results = await super().fetch(*tileables)
../../../deploy/oscar/session.py:866: in fetch
    result.append(self._process_result(tileable, merged))
../../../deploy/oscar/tests/session.py:69: in _process_result
    self.assert_object_consistent(tileable, result)
../../../tests/core.py:357: in assert_object_consistent
    self.assert_series_consistent(expected, real)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = <mars.deploy.oscar.tests.session.CheckedSession object at 0x7f284544fb50>, expected = Series(op=DataFrameIlocGetItem)
real = 0     0.206287
1     0.881219
2     0.354464
3     0.832302
4     0.316828
5     0.205301
6     0.778543
7     0.42978...83043
26    0.826312
27    0.575012
28    0.435271
29    0.059423
30    0.026819
31    0.124065
Name: 1, dtype: float64

    def assert_series_consistent(self, expected, real):
        series_types = (pd.Series,)
        if cudf is not None:
            series_types += (cudf.Series,)
    
        if not isinstance(real, series_types):
            raise AssertionError(f'Type of real value ({type(real)}) not Series')
        self.assert_shape_consistent(expected.shape, real.shape)
    
        if self._check_options['check_series_name'] and expected.name != real.name:
>           raise AssertionError(f'series name in metadata {expected.name} '
>                                          f'is not equal to real name {real.name}')
>           E           AssertionError: series name in metadata None is not equal to real name 1